### PR TITLE
Cherry-pick other changes needed for transformer_4.38 branch.

### PR DESCRIPTION
**Pull request by jiminha**
Cherry-pick "disable sdpa" to transformer_4.38 branch.

### CODE REVIEW

**Comment by jiminha** [comment]
Added one more commit : Move grad.item() inside log method to fix perf degradation.

Below commit is causing perf degration(10-20%) for all models due to gradient copy from HPU to host. Moving inside the log method. 
https://github.com/huggingface/transformers/commit/4f09d0fd888dbf2660313f9715992822acfb99ce

**Comment by regisss** [comment]
Just to be sure, for RoPE with Llama this consists in reverting to Transformers 4.37 right?

**Comment by libinta** [comment]
> Just to be sure, for RoPE with Llama this consists in reverting to Transformers 4.37 right?

yes, use the cos sin cache.
We saw perf drop when not using cache and also accuracy issue now

File: examples/contrastive-image-text/clip_media_pipe.py [hunk 1]
```diff
@@ -128,7 +128,9 @@ def __next__(self):
     read_image_text_from_dataset,
     dtype.NDT,
 )
-op_class = fn.operator_add("ClipDataReader")
+#TODO : NEED to add version check and have backward compatible
+op_class = fn.operator_add("ClipDataReader", False)
+#op_class = fn.operator_add("ClipDataReader")
```
Comment (jiminha) [inline examples/contrastive-image-text/clip_media_pipe.py:133]: @regisss  Habana FW api changes causing the error for bridgetower.  Need to add version check to have both 1.15.0 and 1.14.0 to work.

Comment (regisss) [reply examples/contrastive-image-text/clip_media_pipe.py:133]: Yep, here is an example of version check: https://github.com/huggingface/optimum-habana/blob/20832c92ce3e9cfe94aeb1797a5533b4be962a6c/examples/text-generation/utils.py#L177

Comment (regisss) [reply examples/contrastive-image-text/clip_media_pipe.py:133]: I'm going to push a commit for it

Comment (regisss) [reply examples/contrastive-image-text/clip_media_pipe.py]: @jiminha WDYT? https://github.com/huggingface/optimum-habana/pull/776/commits/8bbe1173581e27f18f1d052472e6a1f4d26bb888

File: optimum/habana/transformers/models/llama/modeling_llama.py [hunk 2]
```diff
@@ -274,17 +299,27 @@ def pre_attn_forward(
...
         key_states = key_states.view(bsz, q_len, -1, self.head_dim).transpose(1, 2)
         value_states = value_states.view(bsz, q_len, -1, self.head_dim).transpose(1, 2)
-
-        past_key_value = getattr(self, "past_key_value", past_key_value)
-
-        cos, sin = self.rotary_emb(value_states, position_ids)
+  
+        kv_seq_len = key_states.shape[-2]
+        # Why do we need this if we pass in?
+        #past_key_value = getattr(self, "past_key_value", past_key_value)
```
[the remark this answers is not in the document]
Comment (libinta) [reply optimum/habana/transformers/models/llama/modeling_llama.py]: ok to remove

**Review by regisss — APPROVED** [review]

File: examples/contrastive-image-text/clip_media_pipe.py (hunk 1 above)
Comment (jiminha) [reply examples/contrastive-image-text/clip_media_pipe.py]: Validated both on 15.0 and 14.0.  No error seen.

**Pull request merged**